mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_id_token.go

```diff
@@ -0,0 +1,161 @@
+	TargetServiceAccount types.String `tfsdk:"target_service_account"`
+	Delegates            types.Set    `tfsdk:"delegates"`
+	IncludeEmail         types.Bool   `tfsdk:"include_email"`
+	IdToken              types.String `tfsdk:"id_token"`
+}
+
+func (p *googleEphemeralServiceAccountIdToken) Schema(ctx context.Context, req ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema.Description = "This ephemeral resource provides a Google OpenID Connect (oidc) id_token."
+	resp.Schema.MarkdownDescription = "This ephemeral resource provides a Google OpenID Connect (oidc) id_token."
+
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"target_audience": schema.StringAttribute{
+				Description: "The audience claim for the `id_token`.",
+				Required:    true,
+			},
+			"target_service_account": schema.StringAttribute{
+				Description: "The email of the service account being impersonated.  Used only when using impersonation mode.",
+				Optional:    true,
+				Validators: []validator.String{
+					fwvalidators.ServiceAccountEmailValidator{},
+				},
+			},
+			"delegates": schema.SetAttribute{
+				Description: "Delegate chain of approvals needed to perform full impersonation. Specify the fully qualified service account name.  Used only when using impersonation mode.",
+				Optional:    true,
+				ElementType: types.StringType,
+				Validators: []validator.Set{
+					setvalidator.ValueStringsAre(fwvalidators.ServiceAccountEmailValidator{}),
+				},
+			},
+			"include_email": schema.BoolAttribute{
+				Description: "Include the verified email in the claim. Used only when using impersonation mode.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"id_token": schema.StringAttribute{
+				Description: "The `id_token` representing the new generated identity.",
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func (p *googleEphemeralServiceAccountIdToken) Configure(ctx context.Context, req ephemeral.ConfigureRequest, resp *ephemeral.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	pd, ok := req.ProviderData.(*fwtransport.FrameworkProviderConfig)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *fwtransport.FrameworkProviderConfig, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	// Required for accessing userAgent and passing as an argument into a util function
+	p.providerConfig = pd
+}
+
+func (p *googleEphemeralServiceAccountIdToken) Open(ctx context.Context, req ephemeral.OpenRequest, resp *ephemeral.OpenResponse) {
+	var data ephemeralServiceAccountIdTokenModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	targetAudience := data.TargetAudience.ValueString()
+	creds := fwtransport.GetCredentials(ctx, fwmodels.ProviderModel{}, false, &resp.Diagnostics)
```

Comment:
This is gross, but necessary to make sure the framework version of GetCredentials receives the data it needs:

```suggestion
	model := fwmodels.ProviderModel{
		Credentials:                        p.providerConfig.Credentials,
		AccessToken:                        p.providerConfig.AccessToken,
		ImpersonateServiceAccount:          p.providerConfig.ImpersonateServiceAccount,
		ImpersonateServiceAccountDelegates: p.providerConfig.ImpersonateServiceAccountDelegates,
		Project:                            p.providerConfig.Project,
		BillingProject:                     p.providerConfig.BillingProject,
		Scopes:                             p.providerConfig.Scopes,
		UniverseDomain:                     p.providerConfig.UniverseDomain,
	}
	creds := fwtransport.GetCredentials(ctx, model, false, &resp.Diagnostics)
```

fwmodels.ProviderModel is the struct that is populated with data from the provider block when the provider is being configured. The fwtransport.GetCredentials function is written in a way that makes it very coupled with being used in the context of a provider being configured, whereas the SDK version of GetCredentials is a method on the Config struct and can be run outside the context of a provider being configured.

Just more evidence that the original muxing was poorly-planned and 💩

Reply:
This is unfortunate :/ , I added a TODO comment so i can go back and fix this.